*******************************************************************************
Date         : 2019-06-20
Author       : Yi Wang
Harman International Industries, Incorporated

Description  : This patch is for fixing mtp device with sd card can't be detected issue
ChangeList   : 5403345
*******************************************************************************

diff --git a/src/simple-mtpfs-mtp-device.cpp b/src/simple-mtpfs-mtp-device.cpp
index a49e9d6..0b35f81 100755
--- a/src/simple-mtpfs-mtp-device.cpp
+++ b/src/simple-mtpfs-mtp-device.cpp
@@ -292,11 +292,15 @@ const TypeDir *MTPDevice::dirFetchContentFromCache(std::string path)
     LogDebug("path:%s", path.c_str());
 
     if (!m_root_dir.isFetched()) {
+        int count = 0;
         for (LIBMTP_devicestorage_t *s = m_device->storage; s; s = s->next) {
             m_root_dir.addDir(TypeDir(s_root_node, 0, s->id, std::string(s->StorageDescription)));
             LogDebug("path:%s setFetched", path.c_str());
             m_root_dir.setFetched();
+            count++;
         }
+        m_root_dir.setItemCount(count);
+        m_root_dir.setFetchHandleDone(true);
     }
 
     if (m_root_dir.dirCount() == 1)
